rust/shalom_core/src/schema/resolver.rs

```diff
@@ -149,17 +151,54 @@ fn resolve_enum(context: SharedSchemaContext, name: String, origin: Node<apollo_
     context.add_enum(name.clone(), Node::new(enum_type));
 }
 
-pub fn resolve_type(_context: SharedSchemaContext, origin: apollo_schema::Type) -> FieldType {
+fn resolve_input(
+    context: &SharedSchemaContext,
+    name: String,
+    origin: &apollo_schema::InputObjectType,
+) {
+    if context.get_type(&name).is_some() {
+        return;
+    }
+    let mut fields = HashMap::new();
+    for (name, field) in origin.fields.iter() {
+        let description = field.description.as_ref().map(|v| v.to_string());
+        let ty = resolve_type(context, &field.ty);
+        let gql_ty = context.get_type(&ty.name()).unwrap();
+        assert!(
+            matches!(gql_ty, GraphQLAny::Scalar(_)) | matches!(gql_ty, GraphQLAny::InputObject(_)),
+            "other input types have not been implemented"
+        );
```

Comment:
ig you need `gql_ty` for the assert, though it can get recursive.. we can do the assert elsewhere.

Reply:
what?